draft-ietf-ppm-dap.md

```diff
@@ -2155,64 +2155,48 @@ aggregation parameter in advance. For example, for Prio3 the only valid
 job is handled asynchronously.
 
-Upon receipt of a `CollectionReq`, the Leader begins by checking that it
+Upon receipt of a `CollectionJobReq`, the Leader begins by checking that it
 recognizes the task ID in the request path. If not, it MUST abort with error
 `unrecognizedTask`.
 
 The Leader MAY further validate the request according to the requirements in
 {{batch-validation}} and abort with the indicated error, though some conditions
 such as the number of valid reports may not be verifiable while handling the
-CollectionReq message, and the batch will have to be re-validated later on
+`CollectionJobReq` message, and the batch will have to be re-validated later on
 regardless.
 
-If the Leader finds the CollectionReq to be valid, it immediately responds with
-HTTP status 201.
-
-The Leader then begins working with the Helper to aggregate the reports
-satisfying the query (or continues this process, depending on the VDAF) as
-described in {{aggregate-flow}}.
-
 Changing a collection job's parameters is illegal, so further requests to
-`PUT /tasks/{tasks}/collection_jobs/{collection-job-id}` for the same
-`collection-job-id` but with a different `CollectionReq` in the body MUST fail
-with an HTTP client error status code.
-
-After receiving the response to its `CollectionReq`, the Collector makes an
-HTTP GET request to
-`{leader}/tasks/{task-id}/collection_jobs/{collection-job-id}` to check on the
-status of the collect job and eventually obtain the result. If the collection job is not
-finished yet, the Leader responds with HTTP status 202 Accepted. The response
-MAY include a Retry-After header field to suggest a polling interval to the
-Collector.
-
-Asynchronously from any request from the Collector, the Leader attempts to run
-the collection job. It first checks whether it can construct a batch for the
-collection job by applying the requirements in {{batch-validation}}. If so, then
-the Leader obtains the Helper's aggregate share following the aggregate-share
-request flow described in {{collect-aggregate}}. If not, it either aborts the
-collection job or tries again later, depending on which requirement in
-{{batch-validation}} was not met. If the Leader has a pending aggregation job
-that overlaps with the batch and aggregation parameter for the collection job,
-the Leader MUST first complete the aggregation job before proceeding and
-requesting an aggregate share from the Helper. This avoids a race condition
-between aggregation and collection jobs that can yield trivial batch mismatch
-errors.
-
-Once both aggregate shares are successfully obtained, the Leader responds to
-subsequent HTTP GET requests to the collection job with HTTP status code 200 OK
-and a body consisting of a `Collection`:
+`PUT /tasks/{task-id}/collection_jobs/{collection-job-id}` for the same
+`collection-job-id` but with a different `CollectionJobReq` in the body MUST
+fail with an HTTP client error status code.
```

Comment:
Typically we would write something like: "the Leader MUST abort with error blah", where "blah" is an error listed in https://ietf-wg-ppm.github.io/draft-ietf-ppm-dap/draft-ietf-ppm-dap.html#table-1. Why not do the same here?

Reply:
I think at some point we decided to leave it unspecified what to return in case of conflict. Aggregation is similar in this respect https://ietf-wg-ppm.github.io/draft-ietf-ppm-dap/draft-ietf-ppm-dap.html#section-4.6.1.2-28.

I don't have an immediate opinion on whether we're right to do this or not. Maybe we should, in case a Leader/Collector needs to take special action based on a conflict response (I think in Janus we just immediately abandon the job).

Reply:
Ack, disregard.